src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,17 +1,25 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{T<:AbstractProjTTN}
+  pm::Vector{T} where {T}
```

Comment:
```suggestion
  pm::Vector{T}
```
You shouldn't need the `where {T}` inside the definition (in fact I don't even know what that does). The way I wrote it, Julia infers that the `T` in the field definition and top level type definition are the same.

Reply:
Another suggestions would be to call this field `terms` instead of `pm`, and then have an accessor function `terms(projected_operator::ProjTTNSum) = projected_operator.terms`. That's a convention I've used for types that represent lazy sums of objects.